utils/yoke/src/yokeable.rs

```diff
@@ -71,16 +71,7 @@ use core::{marker::PhantomData, mem};
 ///     }
 ///
 ///     unsafe fn make(from: Bar<'a>) -> Self {
-///         // We're just doing mem::transmute() here, however Rust is
-///         // not smart enough to realize that Bar<'a> and Bar<'static> are of
-///         // the same size, so instead we use transmute_copy
-///
-///         // This assert will be optimized out, but is included for additional
-///         // peace of mind as we are using transmute_copy
-///         debug_assert!(mem::size_of::<Bar<'a>>() == mem::size_of::<Self>());
-///         let ptr: *const Self = (&from as *const Self::Output).cast();
-///         mem::forget(from);
-///         ptr::read(ptr)
+///         mem::transmute(from)
```

Comment:
Probably worth wrapping with `unsafe {}` for 2024 edition: https://doc.rust-lang.org/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html